### ISSUE TITLE
Miovision - some properly aggregated 15-minute volumes don't have corresponding tmc_uid 

### ISSUE THREAD

**Issue by cczhu** [CityofToronto/bdit_data-sources#410]
Whenever our scripts successfully aggregate 1-minute volumes from `miovision_api.volumes` to 15-minute TMC volumes in `miovision_api.volumes_15min_tmc`, the UID generated for the latter table is written to the former under `volume_15min_tmc_uid` column. If we aggregate all rows in `volumes` where `volume_15min_tmc_uid` is not NULL, we should be able to get the same total volumes as in `volumes_15min_tmc`. We can check this (after ~30 min of calculation) using:

```sql
CREATE TEMPORARY TABLE vol_1min_daily AS (
	SELECT intersection_uid,
	       datetime_bin::date date_bin,
		   classification_uid,
		   SUM(volume) FILTER (WHERE volume_15min_tmc_uid IS NULL) unagg_vol,
		   SUM(volume) FILTER (WHERE volume_15min_tmc_uid IS NOT NULL) agg_vol,
		   SUM(volume) total_vol
	FROM miovision_api.volumes
	WHERE datetime_bin BETWEEN '2020-12-01' AND '2021-06-06 23:59:59'
	GROUP BY intersection_uid, datetime_bin::date, classification_uid
	ORDER BY 1, 2, 3
);

CREATE TEMPORARY TABLE vol_15min_daily AS (
	SELECT intersection_uid,
	       datetime_bin::date date_bin,
		   classification_uid,
		   SUM(volume) total_vol
	FROM miovision_api.volumes_15min_tmc
	WHERE datetime_bin BETWEEN '2020-12-01' AND '2021-06-06 23:59:59'
	GROUP BY intersection_uid, datetime_bin::date, classification_uid
	ORDER BY 1, 2, 3
);
```

The first 10 lines of this query:

```sql
SELECT date_bin,
       intersection_uid,
       classification_uid,
	   a.unagg_vol,
	   a.agg_vol,
	   a.total_vol,
	   b.total_vol total_vol_15mintmc
FROM vol_1min_daily a
LEFT JOIN vol_15min_daily b USING (intersection_uid, date_bin, classification_uid)
WHERE a.agg_vol != b.total_vol
ORDER BY 1, 2, 3
```

are

date_bin | intersection_uid | classification_uid | unagg_vol | agg_vol | total_vol | total_vol_15mintmc
-- | -- | -- | -- | -- | -- | --
2020-12-31 | 1 | 1 | 663 | 18330 | 18993 | 18993
2020-12-31 | 1 | 2 | 1 | 360 | 361 | 361
2020-12-31 | 1 | 3 | 17 | 552 | 569 | 569
2020-12-31 | 1 | 4 | 6 | 352 | 358 | 358
2020-12-31 | 1 | 6 | 24 | 1974 | 1998 | 1998
2020-12-31 | 1 | 7 | 2 | 56 | 58 | 58
2020-12-31 | 2 | 1 | 2433 | 25864 | 28297 | 26862
2020-12-31 | 2 | 2 | 31 | 260 | 291 | 267
2020-12-31 | 2 | 3 | 64 | 732 | 796 | 753
2020-12-31 | 2 | 4 | 54 | 408 | 462 | 411

where `unagg_vol` are total volumes from rows where `volume_15min_tmc_uid IS NULL`, and `agg_vol`  where they are not `NULL`. This shows that in a number of cases, the `agg_vol` is less than the volume from the 15-min TMC table, and only the `total_vol` is equal.

```sql

```
SELECT DISTINCT date_bin
FROM vol_1min_daily a
LEFT JOIN vol_15min_daily b USING (intersection_uid, date_bin, classification_uid)
WHERE a.agg_vol != b.total_vol
```

returns just three days:

![image](https://user-images.githubusercontent.com/442891/121251984-78e5ff80-c875-11eb-8989-905cc0d7e2e2.png)

Counting the number of rows in each hour of each of the three days:

```sql
SELECT DATE_TRUNC('hour', datetime_bin) datetime_bin, COUNT(*) n_rows
FROM miovision_api.volumes WHERE datetime_bin::date IN ('2020-12-31', '2021-01-01', '2021-05-04')
    AND volume_15min_tmc_uid IS NULL AND classification_uid != 7
GROUP BY DATE_TRUNC('hour', datetime_bin);
```

we get

datetime_bin | n_rows
-- | --
2020-12-31 01:00:00 | 1
2020-12-31 10:00:00 | 794
2020-12-31 11:00:00 | 646
2020-12-31 12:00:00 | 723
2020-12-31 14:00:00 | 1
2020-12-31 23:00:00 | 17686
2021-01-01 10:00:00 | 468
2021-01-01 11:00:00 | 141
2021-01-01 14:00:00 | 571
2021-01-01 15:00:00 | 176
2021-01-01 18:00:00 | 1
2021-01-01 19:00:00 | 1
2021-01-01 20:00:00 | 1
2021-01-01 22:00:00 | 5
2021-01-01 23:00:00 | 10200
2021-05-04 16:00:00 | 1554
2021-05-04 17:00:00 | 457
2021-05-04 23:00:00 | 12428

We see huge spikes at 23:00, and more minor ones in the middle of the day. The latter can be explained by unacceptable gaps at a few intersections (though it does look like we might have an inclusive/exclusive timestamp error, since some volumes in the next hour are also eliminated). This can be checked using:

```sql
SELECT * FROM miovision_api.unacceptable_gaps where gap_start::date IN ('2020-12-31', '2021-01-01', '2021-05-04')
ORDER BY 2
```

*though we should also double-check that this is truly the case.*

In conclusion: there's some kind of issue in Miovision aggregation that occasionally aggregates the entire first hour of data (23:00 previous day - 00:00) but does not set the corresponding `volume_15min_tmc_uid` row in `volumes`. It's not obvious from a scan of the code what this could be, but we should keep an eye out for future instances of this.
